components/experimental/src/duration/options.rs

```diff
@@ -0,0 +1,201 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+//! Options for configuring [`DurationFormatter`](crate::duration::DurationFormatter).
+
+/// A bag of options for defining how to format duration using [`DurationFormatter`](crate::duration::DurationFormatter).
+#[derive(Debug, Default, Clone, Copy, PartialEq, Eq)]
+pub struct DurationFormatterOptions {
+    base: BaseStyle,
+
+    year: (YearStyle, UnitVisibility),
+    month: (MonthStyle, UnitVisibility),
+    week: (WeekStyle, UnitVisibility),
+    day: (DayStyle, UnitVisibility),
+    hour: (HourStyle, UnitVisibility),
+    minute: (MinuteStyle, UnitVisibility),
+    second: (SecondStyle, UnitVisibility),
+    millisecond: (MilliSecondStyle, UnitVisibility),
+    microsecond: (MicroSecondStyle, UnitVisibility),
+    nanosecond: (NanoSecondStyle, UnitVisibility),
```

Comment:
This seems more ergonomic than having different wrapper structs for each unit, like:
```rs
struct MinuteFormat {
    style: MinuteStyle,
    visibility: UnitVisibility
}
```
Should I instead prefer the above?

Reply:
We can use `struct` for each option such as `Year`, `Month` ... etc.

and you can add a separate file for each struct with its eums.

Reply:
The options are disjoint in the spec. You can specify `style` and `display`/`visibility` separately.

Reply:
You're right; changed.